mne/channels/montage.py

```diff
@@ -725,6 +726,41 @@ def _get_montage_in_head(montage):
         return transform_to_head(montage.copy())
 
 
+def _set_montage_fnirs(info, montage):
+    """Set the montage for fNIRS data.
+
+    This needs to be different to electrodes as each channel has three
+    coordinates that need to be set. For each channel there is a source optode
+    location, a detector optode location, and a channel midpoint that must be
+    stored. This function modifies info['chs'][#]['loc'] and info['dig'] in
+    place.
+    """
+    # Modify info['chs'][#]['loc'] in place
+    num_ficiduals = len(montage.dig) - len(montage.ch_names)
+    picks = _picks_to_idx(info, 'fnirs', exclude=[], allow_empty=True)
+    info._check_nirs_ch_names()
+    for ch_idx in picks:
+        ch = info['chs'][ch_idx]['ch_name']
+        name, suffix = ch.split(' ')
+        if suffix not in ['hbo', 'hbr']:
+            info['chs'][ch_idx]['loc'][9] = int(suffix)
```

Comment:
This will make it work with intensity or OD if I'm not wrong @rob-luke, is that right?

Reply:
I don't know why the montage setting function would be changing information in the data about light wavelength. That doesn't seem intuitive to me.

Instead you could check that the data structure is correct as is done in the other fNIRS functions and throw an error if its wrong.

See 
https://github.com/mne-tools/mne-python/blob/2a8d56b453ccf4e7b3b22a80bdd42d4ac604474e/mne/preprocessing/nirs/_optical_density.py#L31

or
https://github.com/mne-tools/mne-python/blob/2a8d56b453ccf4e7b3b22a80bdd42d4ac604474e/mne/preprocessing/nirs/_beer_lambert_law.py#L37-L38

Reply:
Except for hb data it would be 

https://github.com/mne-tools/mne-python/blob/70709f5069ddb413c2719f4d2b41fd6bfea2349a/mne/preprocessing/nirs/_beer_lambert_law.py#L59-L60